File: src/main/java/pl/training/workshop/m6/s03_typecode/step3/ScreeningCsv.java
package pl.training.workshop.m6.s03_typecode.step3;

/** Krok 3: CSV rozmawia z mapperem, reszta kodu wyłącznie z typem Format. */
public final class ScreeningCsv {
    /** Wiersz "tytuł;kod" - np. "Diuna;3". */
    public String describe(String line) {
        String[] parts = line.split(";");
        String title = parts[0];
        Format format = FormatCodes.fromCode(Integer.parseInt(parts[1].trim()));
        return title + "|" + format.label() + "|" + format.basePrice()
                + "|okulary:" + (format.requiresGlasses() ? "tak" : "nie")
                + "|csv=" + title + ";" + FormatCodes.toCode(format);
    }
}

File: csharp/src/Training.Workshop/M6/S03TypeCode/Step3/ScreeningCsv.cs
using System.Globalization;

namespace Training.Workshop.M6.S03TypeCode.Step3;

/// <summary>Krok 3: CSV rozmawia z mapperem, reszta kodu wyłącznie z typem Format.</summary>
public sealed class ScreeningCsv
{
    /// <summary>Wiersz "tytuł;kod" - np. "Diuna;3".</summary>
    public string Describe(string line)
    {
        var parts = line.Split(';');
        var title = parts[0];
        var format = FormatCodes.FromCode(int.Parse(parts[1].Trim(), CultureInfo.InvariantCulture));
        return title + "|" + format.Label + "|" + format.BasePrice
            + "|okulary:" + (format.RequiresGlasses ? "tak" : "nie")
            + "|csv=" + title + ";" + FormatCodes.ToCode(format);
    }
}
